go/plugins/firebase/firebase.go

```diff
@@ -22,87 +22,83 @@ import (
 	"log"
 	"sync"
 
-	firebase "firebase.google.com/go/v4"
-	"firebase.google.com/go/v4/auth"
+	firebasev4 "firebase.google.com/go/v4"
 	"github.com/firebase/genkit/go/ai"
 	"github.com/firebase/genkit/go/genkit"
 )
 
-var state struct {
-	mu         sync.Mutex     // Ensures thread-safe access to state
-	initted    bool           // Tracks if the plugin has been initialized
-	app        *firebase.App  // Holds the Firebase app instance
-	retrievers []ai.Retriever // Holds the list of initialized retrievers
+// The provider used in the registry.
+const provider = "firebase"
+
+var appState struct {
```

Comment:
Please follow the pattern in the other plugins, where state is part of the plugin struct. There should be no global state at package-level.

Reply:
I think the reason for this is because Go generally doesn't like global state and things are managed directly right in the main code path. I think we should probably do the same. We should expect that a user will create their own `App` and pass it to the plugin.